src/plopp/graphics/colormapper.py

```diff
@@ -177,56 +178,69 @@ def autoscale(self):
 
-        if self.vmin >= self.vmax:
+        if self._vmin >= self._vmax:
             if self.user_vmax is not None:
-                self.vmax = self.user_vmax
-                self.vmin = self.user_vmax - abs(self.user_vmax) * 0.1
+                self._vmax = self.user_vmax
+                self._vmin = self.user_vmax - abs(self.user_vmax) * 0.1
             else:
-                self.vmin = self.user_vmin
-                self.vmax = self.user_vmin + abs(self.user_vmin) * 0.1
+                self._vmin = self.user_vmin
+                self._vmax = self.user_vmin + abs(self.user_vmin) * 0.1
 
+        self.apply_limits()
+
+    def apply_limits(self):
         # Synchronize the underlying normalizer limits to the current state.
         # Note that the order matters here, as for a normalizer vmin cannot be set above
         # the current vmax.
-        if self.vmin >= self.normalizer.vmax:
-            self.normalizer.vmax = self.vmax
-            self.normalizer.vmin = self.vmin
+        if self._vmin >= self.normalizer.vmax:
+            self.normalizer.vmax = self._vmax
+            self.normalizer.vmin = self._vmin
         else:
-            self.normalizer.vmin = self.vmin
-            self.normalizer.vmax = self.vmax
+            self.normalizer.vmin = self._vmin
+            self.normalizer.vmax = self._vmax
 
-        self._set_artists_colors()
         if self.colorbar is not None:
             self._update_colorbar_widget()
+        self.notify_artists()
 
-    def _set_artists_colors(self):
+    def notify_artists(self):
+        """
+        Notify the artists that the state of the colormapper has changed.
         """
-        Update the colors of all the artists apart from the one that triggered the
-        update, as those get updated by the figure.
+        for artist in self.artists.values():
+            artist.notify_artist('colormap changed')
 
-        Parameters
-        ----------
-        keys:
-            List of artists to update.
+    @property
+    def vmin(self) -> float:
         """
-        for k in self.artists.keys():
-            self.artists[k].set_colors(self.rgba(self.artists[k].data))
+        Get or set the minimum value of the colorbar.
+        """
+        return self._vmin
+
+    @vmin.setter
+    def vmin(self, vmin: sc.Variable | float):
+        self._vmin = maybe_variable_to_number(vmin, unit=self._unit)
+        self.apply_limits()
 
-    def update(self):
+    @property
+    def vmax(self) -> float:
         """
-        Update the colors of all artists, if the `self.set_colors_on_update` attribute
-        is set to `True`.
+        Get or set the maximum value of the colorbar.
         """
-        if self.set_colors_on_update:
-            self._set_artists_colors()
+        return self._vmax
+
+    @vmax.setter
+    def vmax(self, vmax: sc.Variable | float):
+        self._vmax = maybe_variable_to_number(vmax, unit=self._unit)
+        self.apply_limits()
```

Comment:
Given that these properties trigger a computation when they are set, does it make sense to add a method that sets both limits? This could avoid one call to `apply_limits` when both are changed. Basically, I'm asking how often this case happens.

Reply:
For now, I don't think this is actually ever used. When I was implementing, I thought it could be useful to manually set the limits on the colormapper after the figure is created, but we currently have no such examples in practice, the docs or even tests.

Maybe I should just remove the `vmin` and `vmax` properties for now, and only add them when we have a need for them? We will just rely on kwargs from the user to set the limits when the plot is first created.

Reply:
I am fine with leaving them or removing them. If you don't see a use case in the (near) future, maybe just remove them.